### ISSUE TITLE
Re-eval SessionManager/Session/LiveData; consider kotlinify

### ISSUE THREAD

**Issue by mcomella** [mozilla-mobile/firefox-tv#415]
I always found LiveData a little weird and hard to follow - should we replace it? Perhaps the structure of SessionManager and SessionCallbackProxy is the root of the problems?

**Comment by liuche** [comment]
Cleaning up titles so [meta] tag is clearer.

**Comment by mcomella** [comment]
We said less important than 2.0 work.

**Comment by mcomella** [comment]
We don't use tabs and it hasn't gotte in the way yet - let's just close it.
